api/app/models/env_canada.py

```diff
@@ -154,74 +167,171 @@ def download(url: str, path: str) -> str:
     return target
 
 
-def flag_file_as_processed(session, url):
-    """ Flag the file as processed in the database """
-    processed_file = get_processed_file_record(session, url)
-    if processed_file:
-        logger.info('re-procesed %s', url)
-    else:
-        logger.info('file processed %s', url)
-        processed_file = ProcessedModelRunUrl(
-            url=url,
-            create_date=datetime.datetime.now(datetime.timezone.utc))
-    processed_file.update_date = datetime.datetime.now(datetime.timezone.utc)
-    session.add(processed_file)
-    session.commit()
+def mark_prediction_model_run_processed(session: Session,
+                                        model: ModelEnum,
+                                        projection: str,
+                                        now: datetime.datetime,
+                                        hour: int):
+    """ Mark a prediction model run as processed (complete) """
 
+    prediction_model = app.db.crud.get_prediction_model(
+        session, model, projection)
+    prediction_run_timestamp = datetime.datetime(
+        year=now.year,
+        month=now.month,
+        day=get_model_day(now, hour),
+        hour=hour, tzinfo=datetime.timezone.utc)
+    logger.info('prediction_model:%s, prediction_run_timestamp:%s',
+                prediction_model, prediction_run_timestamp)
+    prediction_run = app.db.crud.get_prediction_run(
+        session,
+        prediction_model.id,
+        prediction_run_timestamp)
+    prediction_run.complete = True
+    app.db.crud.update_prediction_run(session, prediction_run)
 
-def main():
-    """ main script """
-    start_time = time.time()
-    session = app.db.database.get_session()
-    files_downloaded = 0
-    files_processed = 0
-    exception_count = 0
-    urls = get_download_urls()
-    processor = GribFileProcessor()
-    with tempfile.TemporaryDirectory() as gdps_path:
-
-        for url, filename in urls:
+
+class EnvCanada():
+    """ Class that orchestrates downloading and processing of weather model grib files from environment
+    Canada.
+    """
+
+    def __init__(self):
+        """ Prep variables """
+        self.files_downloaded = 0
+        self.files_processed = 0
+        self.exception_count = 0
+        # We always work in UTC:
+        self.now = get_utcnow()
+        self.session = app.db.database.get_session()
+        self.grib_processor = GribFileProcessor()
+
+    def flag_file_as_processed(self, url):
+        """ Flag the file as processed in the database """
+        processed_file = get_processed_file_record(self.session, url)
```

Comment:
Are you querying this again to update `update_date` field?

Reply:
On it's own this function makes sense - I agree in the context of it's usage, it does a redundant check and log - but I don't mind - the logic is sound by itself.